pyasic/data/pools.py

```diff
@@ -1,4 +1,24 @@
 from dataclasses import dataclass, field
+from enum import Enum
+from typing import Optional
+
+
+class Scheme (Enum):
+    STRATUM_V1 = "stratum+tcp"
+    STRATUM_V2 = "stratum2+tcp"
+
+@dataclass
+class PoolUrl:
+    scheme: Scheme
+    host: str
+    port: int
+    pubkey: Optional[str] = None
+
+    def __str__(self) -> str:
+        if self.scheme == Scheme.STRATUM_V2 and self.pubkey:
+            return f"{self.scheme.value}://{self.host}:{self.port}/{self.pubkey}"
+        else:
+            return f"{self.scheme.value}://{self.host}:{self.port}"
```

Comment:
Maybe want a `from_str` function that parses a string formatted pool URl into this format.  You could likely use something from stdlib like urllib.

Reply:
do you mean I should replace ` __str__(self)` with `from_str`?

Reply:
which format should be the desired output? 

1. stratum2+tcp://btc.pool.com:3333/pubkey
2. PoolUrl(scheme=<Scheme.STRATUM_V2: 'stratum2+tcp'>, host='btc.pool.com', port=3333, pubkey='pubkey')

Reply:
Precisely.

Reply:
okay, I will push an update soon.